File: code/javascript/numberology.js
/**
 * numberology.js by Isaac Halvorson
 *
 * A simple implementation of basic Numerology theory written in JavaScript
 */

function numberology() {

	// Alert the user as to what Numberology is
	alert(
		"Welcome to Numberology!\n\nNumberology is a simple implementation of Numerology theory written in JavaScript by Isaac Halvorson"
	);

	// Prompt the user for required data
	const fullName = prompt("To start things off, I'm going to need your full name (including your middle name). Please type it in here and hit enter:");
	const birthYear = prompt("Ok, now please enter the year in which you were born and hit enter:");
	const birthMonth = prompt("Next, I'll need the month (you can enter the full month name, an abbreviation, or the month's number):");
	const birthDay = prompt("Finally, please enter the day of your birth:");
}

function getCharacterNumber() {

}

numberology();

File: code/python3/numberology.py
# numberology.py by Isaac Halvorson
#
# A Python 3 implementation of simple Numerology algorithms

def calculate_final_number(number_string):
	return (int(number_string) - 1) % 9 + 1


def get_character_number(character):
	if character.isnumeric():
		return character
	elif character.isalpha():
		return ord(character.lower()) - 96
	else:
		return 0


def numberology(text):
	big_number_string = ""
	
	for character in text:
		big_number_string += str(get_character_number(character))
	
	return calculate_final_number(big_number_string)


name = input("Please enter your full name: ")
name_number = numberology(name)
print("Your name number is:", name_number)

birthday = input("Please enter your birthday: ")
life_number = numberology(birthday)
print("Your life number is:", life_number)
